src/Symfony/Component/Notifier/Bridge/Mercure/MercureTransport.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mercure;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Mercure\PublisherInterface;
+use Symfony\Component\Mercure\Update;
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+final class MercureTransport extends AbstractTransport
+{
+    private $publishers;
+    private $publisherId;
+    private $topic;
+
+    /**
+     * @param ContainerInterface $publisherLocator A container that holds {@see PublisherInterface} instances
+     */
+    public function __construct(ContainerInterface $publisherLocator, string $publisherId, ?string $topic = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
```

Comment:
Why don't you inject the Publisher instance directly instead of a service locator? This would be a cleaner design and will help reusing  the component without FrameworkBundle.

Reply:
You're perfectly right.